src/pip/_internal/utils/filesystem.py

```diff
@@ -169,3 +170,62 @@ def _test_writable_dir_win(path):
     raise EnvironmentError(
         'Unexpected condition testing for writable directory'
     )
+
+
+def find_files(path, pattern):
+    # type: (str, str) -> List[str]
+    """Returns a list of absolute paths of files beneath path, recursively,
+    with filenames which match the UNIX-style shell glob pattern."""
+    result = []  # type: List[str]
+    for root, dirs, files in os.walk(path):
+        matches = fnmatch.filter(files, pattern)
+        result.extend(os.path.join(root, f) for f in matches)
+    return result
+
+
+def _friendly_size(size):
+    # type: (Union[float, int]) -> str
+    suffix = 'B'
+    if size > 1000:
+        size /= 1000
+        suffix = 'KB'
+
+    if size > 1000:
+        size /= 1000
+        suffix = 'MB'
+
+    if size > 1000:
+        size /= 1000
+        suffix = 'GB'
+
+    size = round(size, 1)
+
+    return '{} {}'.format(size, suffix)
```

Comment:
This might return a misleading result on Python 2 when `size` is an int (yeah, Python 2…)